snapcraft/internal/meta/_desktop.py

```diff
@@ -35,21 +35,17 @@ def __init__(
         self._prime_dir = prime_dir
         self._path = os.path.join(prime_dir, filename)
         if not os.path.exists(self._path):
-            raise errors.InvalidDesktopFileError(
-                filename, 'does not exist (defined in the app {!r})'.format(
-                    name))
+            raise errors.MissingDesktopFileError
```

Comment:
Wait, no, these were good! You still need the parameters on these exceptions! You just need to ensure they're using variables that exist :) .